### PR TITLE
Exception safe extend

### PR DESCRIPTION
**Pull request by k-morozov**
Possible error without this PR:

```
2024-11-07 01:59:19,569 MainProcess 4116997 [DEBUG   ] chadmin: detect-broken-partitions: Checking files from: /var/lib/clickhouse/disks/object_storage/store/e37/e3756c3a-d3dc-493d-988a-2b7539b9bc81/20241027_1_1_0
2024-11-07 01:59:19,590 MainProcess 4116997 [ERROR   ] chadmin: detect-broken-partitions: Command 'detect-broken-partitions' failed with error:
Traceback (most recent call last):
  File "/opt/yandex/clickhouse-tools/lib/python3.6/site-packages/ch_tools/chadmin/cli/chadmin_group.py", line 51, in wrapper
    cmd_callback(*a, **kw)
  File "/opt/yandex/clickhouse-tools/lib/python3.6/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/opt/yandex/clickhouse-tools/lib/python3.6/site-packages/ch_tools/chadmin/cli/data_store_group.py", line 378, in detect_broken_partitions
    objects.extend(S3ObjectLocalMetaData.from_file(file_full_path).objects)
  File "/opt/yandex/clickhouse-tools/lib/python3.6/site-packages/ch_tools/chadmin/internal/object_storage/s3_object_metadata.py", line 85, in from_file
    if path.stat().st_size > MAX_METADATA_FILE_SIZE:
  File "/usr/lib/python3.6/pathlib.py", line 1158, in stat
    return self._accessor.stat(self)
  File "/usr/lib/python3.6/pathlib.py", line 387, in wrapped
    return strfunc(str(pathobj), *args)
FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/clickhouse/disks/object_storage/store/e37/e3756c3a-d3dc-493d-988a-2b7539b9bc81/20241027_1_1_0/degradation.sparse.idx.cmrk2'
```

### CODE REVIEW

File: ch_tools/chadmin/cli/data_store_group.py [hunk 1]
```diff
@@ -513,9 +516,6 @@ def detach_partition(ctx: Context, table_partition: TablePartition) -> bool:
     Run DETACH the partition.
     """
 
-    logging.debug(
```
Comment (MikhailBurdukov) [inline ch_tools/chadmin/cli/data_store_group.py:516]: why?(

File: ch_tools/chadmin/cli/data_store_group.py [hunk 2]
```diff
@@ -374,8 +374,11 @@ def detect_broken_partitions(ctx, root_path, reattach, detach):
         objects: List[S3ObjectLocalInfo] = []
         logging.debug(f"Checking files from: {path}")
         for file in files:
-            file_full_path = Path(os.path.join(path, file))
-            objects.extend(S3ObjectLocalMetaData.from_file(file_full_path).objects)
+            try:
+                file_full_path = Path(os.path.join(path, file))
+                objects.extend(S3ObjectLocalMetaData.from_file(file_full_path).objects)
+            except Exception as e:
```
Comment (MikhailBurdukov) [inline ch_tools/chadmin/cli/data_store_group.py:380]: Can we handle more specific exception? `Exception` is too general.

And why 
```
 file_full_path = Path(os.path.join(path, file))
```
operation must be in try?

File: ch_tools/chadmin/cli/data_store_group.py (hunk 1 above)
Comment (k-morozov) [reply ch_tools/chadmin/cli/data_store_group.py:516]: query is being already written by query_with_retry.

File: ch_tools/chadmin/cli/data_store_group.py (hunk 2 above)
Comment (k-morozov) [reply ch_tools/chadmin/cli/data_store_group.py:380]: What problem will be resolved if we use a more specific exception?

Comment (MikhailBurdukov) [reply ch_tools/chadmin/cli/data_store_group.py:380]: For example, a broken S3 connection will result in overloaded logs, a lot of junky s3 requests. 
And we also can't imagine all such cases, so better to handle only what we already know.
We can use `keep-going` if you want to ignore all exceptions.

Comment (k-morozov) [reply ch_tools/chadmin/cli/data_store_group.py:380]: If it is a problem what quantities of extra logs/requests are we talking about?

**Review by aalexfvk — APPROVED** [review]

**Pull request merged**